### PR TITLE
concentrated-liquidity: add test to check (de)serialization of cl's messages

### PR DESCRIPTION
**Pull request by RusAkh**
- go work sum
- MsgCreateConcentratedPool (de)ser check
- test for MsgCreateConcentratedPool
- MsgWithdrawPosition check
- MsgWithdrawPosition test case
- MsgCreatePosition check
- MsgCreatePosition test case
- add check to make sure result is expected and add else branch for msgs not yet checked

Closes: #3914

## What is the purpose of the change

adds test that makes sure serialization/deserialization of CL's messages work as expected

## Documentation and Release Note
no

### CODE REVIEW

File: x/concentrated-liquidity/types/msgs_test.go [hunk 1]
```diff
@@ -285,3 +288,53 @@ func TestMsgWithdrawPosition(t *testing.T) {
...
+				TokenDesired0:   sdk.NewCoin("foo", sdk.NewInt(1000)),
+				TokenDesired1:   sdk.NewCoin("bar", sdk.NewInt(1000)),
+				TokenMinAmount0: sdk.OneInt(),
+				TokenMinAmount1: sdk.OneInt(),
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			apptesting.TestMessageConcentratedLiquiditySerialization(t, tc.clMsg)
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/msgs_test.go:337]: I might be messing something but why is not possible to reuse the existing helper like here:
https://github.com/osmosis-labs/osmosis/blob/f611e01c7fb9ab9d5348250cb0b65e08f3ec651b/x/swaprouter/types/msgs_test.go#L323

File: app/apptesting/test_suite.go [hunk 2]
```diff
@@ -396,6 +400,35 @@ func CreateRandomAccounts(numAccts int) []sdk.AccAddress {
 	return testAddrs
 }
 
+func TestMessageConcentratedLiquiditySerialization(t *testing.T, msg sdk.Msg) {
```
Comment (p0mvn) [inline app/apptesting/test_suite.go:403]: Why are we defining concentrating liqudity-specific helpers in the global apptesting package?

Let's keep this appropriately scoped in the CL module please.

To build up on my other comment, I'm not sure why we need this custom helper and why we cannot reuse the one I linked

File: x/concentrated-liquidity/types/msgs_test.go (hunk 1 above)
Comment (RusAkh) [reply x/concentrated-liquidity/types/msgs_test.go:337]: @p0mvn oh, I completely misunderstood the task at the beginning, now I get what is required, I will change the code and remove stuff I wrote to apptesting

Comment (p0mvn) [reply x/concentrated-liquidity/types/msgs_test.go]: No worries. Please tag me once this is ready for another round of review

**Review by p0mvn — APPROVED** [review]

**Pull request merged**